packages/react-form-state/src/validators.ts

```diff
@@ -133,6 +133,32 @@ export function validate<Input, Fields = never>(
   };
 }
 
+export function validateRequired<Input>(
+  matcher: Matcher<Input>,
+  errorContent: ErrorContent,
+): (input: Input) => ErrorContent | void;
+
+export function validateRequired<Input, Fields = never>(
```

Comment:
I run into the same type of typing issue as https://github.com/Shopify/quilt/pull/349

Otherwise this looks good!